components/suggestion/at_mention_provider/at_mention_suggestion.tsx

```diff
@@ -17,15 +17,32 @@ import Suggestion from '../suggestion.jsx';
 import CustomStatusEmoji from 'components/custom_status/custom_status_emoji';
 import StatusIcon from 'components/status_icon';
 
+interface Item {
+    is_bot: any;
+    remote_id: any;
+    status: string;
+    last_picture_update: number;
+    display_name: any;
+    name: string;
+    isCurrentUser: any;
+    type: string;
+    id: string;
+    username: string;
+    nickname: string;
+    first_name: string;
+    last_name: string;
+    roles: string;
+}
+
 export default class AtMentionSuggestion extends Suggestion {
     render() {
-        const isSelection = this.props.isSelection;
-        const item = this.props.item;
+        const isSelection: true | false = this.props.isSelection;
+        const item: Item = this.props.item;
 
-        let itemname;
-        let description;
-        let icon;
-        let customStatus;
+        let itemname: string;
+        let description: JSX.Element | string | null = null;
```

Comment:
We can use `React.ReactNode` for these types instead of adding a bunch of our own.